docs/keystone-auth.md

```diff
@@ -0,0 +1,38 @@
+An authentication backend for Openstack's Keystone is maintained in this
+source tree as `haas.ext.auth.keystone`. This document describes its
+usage.
+
```

Comment:
AFAIK: For the backend we are supporting whatever Keystone Middleware is supporting, and for the CLI, Identity API v3.